tracks/ruby/exercises/high-scores/mentoring.md

```diff
@@ -0,0 +1,71 @@
+New concepts: Array query methods (+ chaining methods), instantiating a class, getter method or `attr_reader`.
+
+### Minimal solution for approval
+
+```ruby
+
+# version 2+
+
+class HighScores
+  def initialize(scores)
+    @scores = scores
+  end
+
+  def scores  # no need to change getter method to `attr_reader`
+    @scores
+  end
+
+  def personal_best
+    scores.max
+  end
+
+  def latest
+    scores.last
+  end
+
+  def personal_top
+    scores.sort.reverse.take(3) 
+  end
+
+  # composing a string either with string interpolation or with an if.. else.. statement
+  def report
+    difference = "#{personal_best - latest} short of" unless personal_best == latest
+
+    "Your latest score was #{latest}. That's #{difference} your personal best!".squeeze
+  end
+end
+
+```
+
+### Reasonable approaches
+
+- See the one above
+- Defining `@scores`, `@personal_best` and `personal_top` in the constructor.  
+
+### General notes
+- The Instructions point to a beginner friendly explanation of instantiating a class: 
+http://ruby-for-beginners.rubymonstas.org/writing_classes/initializers.html
+and to the rubydocs Array. 
+- Most submissions use an getter method `scores`, not `attr_reader`. 
+That's okay at this point in the track, no need to suggest `attr_reader`.
+- Beginners create the attribute reader, but may call the instance variable instead of the getter. 
+This link gives a nice introduction of getters and setters, and the naming convention in Ruby:
+https://launchschool.com/books/oo_ruby/read/classes_and_objects_part1#accessormethods.
+It's recommended to link to this or your favorite explanation, instead of typing the solution for them.
+- `sort.reverse` is not very performant, but at this point in the track it's more important that students explore the 
+standard lib (and work with methods chaining) than that they know how to optimize.
```

Comment:
"standard lib" should be "core" and "library".  No need to abbreviate a word, and especially not one that comes up with definitions that may not be clear in the English dictionary.  Since this is Ruby, there is a difference between "core" and "standard" and often nothing more than core is necessary.  Though I agree it is important that students explore both collections.

Reply:
I use 'Core' and 'Standard' when referring to specific Ruby libraries. With lower case 'standard' I meant it in a more general way.

Reply:
Easy to [stroop](https://faculty.washington.edu/chudler/words.html) though.